aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -267,6 +267,42 @@ std::tuple<Tensor, Tensor> cummax(const Tensor& self, int64_t dim) {
   at::cummax_out(values, indices, self, dim);
   return std::tuple<Tensor &,Tensor &>{values, indices};
 }
+
+std::tuple<Tensor&, Tensor&> cummin_out(Tensor& values, Tensor& indices, const Tensor& self, int64_t dim) {
+  check_scalar_type_device_layout_equal(values, self);
+  check_scalar_type_device_layout_equal(indices, at::empty({0}, self.options().dtype(at::kLong)));
+  {
+    NoNamesGuard guard;
+    values.resize_(self.sizes());
+    indices.resize_(self.sizes());
+    if(self.dim() == 0) {
+      values.fill_(self.item());
```

Comment:
do you need to call `item` on self here?  Don't we have an overload that takes a tensor?

Reply:
oh true! updated